agogosml_cli/cli/templates/{{cookiecutter.PROJECT_NAME_SLUG}}/e2e/testgen/main.py

```diff
@@ -5,19 +5,65 @@
 from multiprocessing.pool import ThreadPool
 
 from agogosml.common.eventhub_streaming_client import EventHubStreamingClient
+from agogosml.common.kafka_streaming_client import KafkaStreamingClient
+
+eh_base_config = {
+    "EVENT_HUB_NAMESPACE": os.getenv("EVENT_HUB_NAMESPACE"),
+    "EVENT_HUB_NAME": os.getenv("EVENT_HUB_NAME_INPUT"),
+    "EVENT_HUB_SAS_POLICY": os.getenv("EVENT_HUB_SAS_POLICY"),
+    "EVENT_HUB_SAS_KEY": os.getenv("EVENT_HUB_SAS_KEY_INPUT"),
+}
+
+eh_send_config = {
+    **eh_base_config,
+    'LEASE_CONTAINER_NAME': os.getenv('LEASE_CONTAINER_NAME_INPUT')
+}
+
+eh_receive_config = {
+    **eh_base_config,
+    "AZURE_STORAGE_ACCOUNT": os.getenv("AZURE_STORAGE_ACCOUNT"),
+    "AZURE_STORAGE_ACCESS_KEY": os.getenv("AZURE_STORAGE_ACCESS_KEY"),
+    "LEASE_CONTAINER_NAME": os.getenv("LEASE_CONTAINER_NAME_OUTPUT"),
+    "EVENT_HUB_CONSUMER_GROUP": os.getenv("EVENT_HUB_CONSUMER_GROUP"),
+    "TIMEOUT": 10
+}
+
+kafka_base_config = {
+    'KAFKA_ADDRESS': os.getenv("KAFKA_ADDRESS"),
+    'TIMEOUT': os.getenv('TIMEOUT'),
+    # These configs are specific to Event Hub Head for Kafka
+    'EVENTHUB_KAFKA_CONNECTION_STRING': os.getenv('EVENTHUB_KAFKA_CONNECTION_STRING'),
+    'SSL_CERT_LOCATION': os.getenv('SSL_CERT_LOCATION')
+}
+
+kafka_receive_config = {
+    **kafka_base_config,
+    'KAFKA_CONSUMER_GROUP': os.getenv('KAFKA_CONSUMER_GROUP'),
+    'KAFKA_TOPIC_INPUT': os.getenv('KAFKA_TOPIC_INPUT')
+}
+
+kafka_send_config = {
+    **kafka_base_config,
+    'KAFKA_TOPIC_OUTPUT': os.getenv('KAFKA_TOPIC_OUTPUT')
+}
+
+
+def receive_messages(msg_type: str):
+    pool = ThreadPool(processes=1)
+
+    async_result = pool.apply_async(get_messages_from_client, args=(msg_type,))
 
+    result = async_result.get()
 
-def send_messages():
+    return json.dumps(result)
+
+
+def send_messages_to_client(msg_type: str):
```

Comment:
This looks like it should re-use `agogosml.tools.sender.main` instead. Can probably replace a lot of this script with that, including looking up the correct client based on the config.